sequencer/onchain.go

```diff
@@ -165,21 +167,62 @@ func (s *Sequencer) pushTransitionToContract(processID []byte,
 	); err != nil {
 		log.Debugw("failed to simulate state transition",
 			"error", err,
-			"pid", hex.EncodeToString(processID))
+			"pid", processID.String())
 	}
 
 	// Submit the proof to the contract
-	txHash, err := s.contracts.SetProcessTransition(processID,
+	log.Infow("state transition pending to be mined",
+		"pid", processID.String())
+	if err := s.contracts.SetProcessTransitionAndWait(
+		processID,
 		abiProof,
 		abiInputs,
 		blobSidecar,
-	)
-	if err != nil {
-		return fmt.Errorf("failed to submit state transition: %w", err)
+		transitionOnChainTimeout,
+		s.pushStateTransitionCallback(processID),
+	); err != nil {
+		if err := s.stg.ClearPendingTx(storage.StateTransitionTx, processID); err != nil {
+			log.Warnw("failed to release pending tx",
+				"error", err,
+				"processID", processID.String())
+		}
+		log.Infow("pending tx released", "pid", processID.String())
 	}
+	return nil
+}
 
-	// Wait for the transaction to be mined
-	return s.contracts.WaitTx(*txHash, time.Minute)
+func (s *Sequencer) pushStateTransitionCallback(pid types.HexBytes) func(err error) {
+	return func(err error) {
+		defer func() {
+			// Remove the pending tx mark
+			if err := s.stg.ClearPendingTx(storage.StateTransitionTx, pid); err != nil {
```

Comment:
"Clear" is not correctly used here, it should be either "Clean", "Prune", "Remove"